File: problems/029-source-tree-evidence/solutions/python/main.py
import sys

l = sys.stdin.buffer.read().splitlines()
n, E = l[0].split()
a = []
for s in l[1:]:
    p = s.split()[1]
    if p != E and not p.startswith(E + b"/"):
        a.append((p, s))
a.sort()
sys.stdout.buffer.write(
    str(len(a)).encode() + b"\n" + b"".join(s + b"\n" for _, s in a)
)

File: problems/029-source-tree-evidence/solutions/typescript/main.ts
import * as std from "std";
const l = std.in.readAsString().trimEnd().split(/\r?\n/),
  h = l[0].split(" "),
  n = +h[0],
  e = h[1],
  a: [string, string][] = [];
for (let i = 1; i <= n; i++) {
  const p = l[i].split(" ")[1];
  if (p !== e && !p.startsWith(e + "/")) a.push([p, l[i]]);
}
a.sort((x, y) => x[0] < y[0] ? -1 : x[0] > y[0] ? 1 : 0);
std.out.puts([`${a.length}`, ...a.map((x) => x[1])].join("\n") + "\n");
